code/API_definitions/age-verification.yaml

```
@@ -0,0 +1,511 @@
+
+  description: |
+
+    # Summary
+    This API provides the customer with the ability to check if the user of the line is older than a provided age, in order to provide API customer's age-restricted services, access to its age-restricted website etc..
+
+    # Authorization and authentication
+
+    The "Camara Security and Interoperability Profile" provides details on how a client requests an access token. Please refer to Identify and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the Profile.
+
+    Which specific authorization flows are to be used will be determined during onboarding process, happening between the API Client and the Telco Operator exposing the API, taking into account the declared purpose for accessing the API, while also being subject to the prevailing legal framework dictated by local legislation.
+
+    It is important to remark that in cases where personal user data is processed by the API, and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of 3-legged access tokens becomes mandatory. This measure ensures that the API remains in strict compliance with user privacy preferences and regulatory obligations, upholding the principles of transparency and user-centric data control.
+
+    # API functionality
+
+    The API defines one service endpoint:
+
+    - `POST /ageverify`
+
+      Takes the network subscription identifier (e.g. the mobile phone number for a mobile network subscriber) and check the age of the user is older than the age's threshold in the request.
+      If information of the user's identity such as `givenName`, `familyName`, `address` match with information held by the MNO.
+
+
+    ## Inputs
+
+    The endpoint request body is a JSON object with the following parameters:
+    - `phoneNumber`: The network subscription identifier (i.e. the phone number of the subscriber). [Required only in case a 2-legged flow has been agreed between API Provider and API Consumer; otherwise, none]
+    - `ageThreshold`: The date's threshold from which the age of the user must be compared. [Required]
+    - any of `idDocument`, `name`, `givenName`, `familyName`, `middleNames`, `familyNameAtBirth`, `birthdate`, `email`; user's information to be checked in order to confirm that the user is the contract's owner [Optional]
+
+    
+
+    ## Outputs
+
+    If successful, a JSON object is returned containing the following data:
+    - `ageCheck`: Indicate 'true' when the age of the user is the same age or older than the age threshold (age >= age threshold), and 'false' if not (age < age threshold).  Otherwise, 'not_available'.
+    - `verifiedStatus`: true if the information provided was checked against another form of official identification, otherwise false.
+    - `identityMatchScore`: the overall match score of identity information provided in the request compared to what MNO holds.

```

Suggestion to add notes to these two sentences for more clarificaiton as below:

   - `verifiedStatus`: true if the information provided was checked against another form of official identification, otherwise false.
   Note: It is optional if an API provider wants to indicate this status.
   - `identityMatchScore`: the overall match score of identity information provided in the request compared to what MNO holds.
   Note: It is optional for an API provider to return the match score.  If an API provider decides not to return the match score, even though any of `idDocument`, `name`, `givenName`, `familyName`, `middieNames`, `familyNameAtBirth`, `birthdate`, `email` are included in the request body, no identityMatchScore parameter will not be returned.

I have commented similar suggestion already https://github.com/camaraproject/KnowYourCustomer/pull/50#issuecomment-2484835842, but it has not been captured yet.

Thanks,
Toshi

---

Hi @ToshiWakayama-KDDI! Thanks for the proposal. From TEF we are thinking in something similar. We would like to complete the text stating that, additionally to what you include about not returning the `identityScore`, it can also be calculated using internal data from the MNO and not only from the input received. For example if the line is part of a multi-line contract so we cannot be sure if it is the owner of the line, if it has parental control activated it may indicate that it is a line used by a child, etc

```suggestion
    - `identityMatchScore`: The overall score of identity information available in the Operator, information either provided in the request body comparing it to the one that the MNO holds or directly using internal MNO's information. It is optional for the Operator to return the Identity match score.

```

---

Sure, we were thinking that there is information not provided in the request that can also be useful to calculate the identity score, for example, let's say that the line is part of a multi-line contract. If you provide the name or idDocument, etc, it will match with the ones owned by the mno, but it won't be precise since the line can be a line used by a child or a spouse. So using internal information can help increase or reduce the match score. Other information can also be useful, for example if the line has any kind of parental control, it can indicate that the data match the owner's but since it has the parental control it may indicate that is used by a child.

I hope this explains better the idea @GillesInnov35